daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -259,10 +262,8 @@ final class ImmArray[+A] private (
     val builder = ImmArray.newBuilder[B]
     var i = 0
     while (i < len) {
-      f.lift(uncheckedGet(i)) match {
-        case None => ()
-        case Some(x) => builder += x
-      }
+      val a = uncheckedGet(i)
```

Comment:
lifting is not very efficient here, it creates too many useless objects in the process